rd-kt/rd-core/src/commonMain/kotlin/com/jetbrains/rd/util/reactive/Signal.kt

```diff
@@ -46,10 +48,18 @@ class Signal<T> : ISignal<T> {
                         if (arr.contains(handler)) throw IllegalArgumentException("Duplicate handler: $handler")
                         arr.insert(handler, arr.size)
                     }
+                    if (queue.get().size == 10_000) {
```

Comment:
Wouldn't it be better to check this inside of `getAndUpdate` callback?
1) it won't require you to get the queue for the second time
2) I have a feeling that this code could easily _skip_ printing a message if there's high contention on an atomic (e.g. if two handlers were added between this and the previous step)

Reply:
Fair point, especially the second one